spark-bigquery-connector-common/src/test/java/com/google/cloud/spark/bigquery/integration/OpenLineageIntegrationTestBase.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2024 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.spark.bigquery.integration;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.cloud.spark.bigquery.integration.SparkBigQueryIntegrationTestBase.TestDataset;
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.SaveMode;
+import org.apache.spark.sql.SparkSession;
+import org.json.JSONObject;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.ExternalResource;
+
+public class OpenLineageIntegrationTestBase {
+  @ClassRule
+  public static TestDataset testDataset = new SparkBigQueryIntegrationTestBase.TestDataset();
+
+  @ClassRule public static CustomSessionFactory sessionFactory = new CustomSessionFactory();
```

Comment:
Why is this class needed? it is used just once so it can be a factory method (calling class can provide the file)

Reply:
I have followed a similar pattern as [SparkBigQueryIntegrationTestBase](https://github.com/GoogleCloudDataproc/spark-bigquery-connector/blob/098a3c26a7dc1630ba584c6e1b149d08ca05d6d4/spark-bigquery-connector-common/src/test/java/com/google/cloud/spark/bigquery/integration/SparkBigQueryIntegrationTestBase.java), with the difference being that this factory initializes both the file and the spark session, whereas in the other class, it is used just for the spark session.